tests/test_scatterPlot.py

```diff
@@ -117,11 +121,40 @@ def test_ScatterPlotWithTwoHistsTask(self):
                                bands=self.bands,
                                plotName="test")
 
+        self.assertTrue(isinstance(result.scatterPlot, plt.Figure))
+
         filename_figure_tmp = os.path.join(self.testDir, "test_scatterPlot.png")
+        texts, lines = get_and_remove_figure_text(result.scatterPlot)
+
+        # Set to true to re-generate reference data
+        resave = False
+
+        for idx, line in enumerate(lines):
+            filename = os.path.join(path_lines_ref, f"line_{idx}.txt")
+            if resave:
+                np.savetxt(filename, line)
+            arr = np.loadtxt(filename)
+            assert np.isclose(arr, line).flat
```

Comment:
Are you sure this is doing what you think it's doing?  I would have thought coercing a `.flat` access to a boolean would not automatically imply something like `np.all`.

You could also just use `assertFloatsAlmostEqual` (from `lsst.utils.tests.TestCase`).

Reply:
I used `assertFloatsAlmostEqual` as suggested, but the default tolerances failed on macOS at the `1e-12` level, which is somewhat concerning (I hadn't run Jenkins before). I set `atol=1e-8, rtol=1e-10` for now, but I'm not sure how robust that's going to be.

Reply:
Given that the result is normally consumed by just human eyes, I'm not particularly bothered by really any loss of precision here (who knows what shenanigans matplotlib might be up to?) and would recommend tolerances around `1E-3`.